### ISSUE TITLE
Loop suddenly ‘forgets’ all therapy settings.

### ISSUE THREAD

**Issue by ChrisVNDL** [LoopKit/Loop#1755]
**Describe the bug**

Hello!

I have built Loop (v2.3.0) from dev branch last sunday and twice now it has suddenly lost all the therapy settings during normal operation.

When this occurs the app starts up and presents like a fresh install. It opens with the welcome screen: 'Welcome to loop'. You can then proceed to configure all over again and it will then function perfectly fine with the existing paired sensor and pump (Dexcom G6 and Omnipod DASH in our case). In other words; it only forget the therapy settings and not the paired hardware it seems.

Both times this has happened nothing out of the ordinary happened; no errors or other events. The Loop app was started and immediately presented with the welcome screen. This seems very odd as it otherwise functions perfectly and there is no indication as to anything causing this behaviour.

**Attach an Issue Report**
[Export-20220721T203916Z.zip](https://github.com/LoopKit/Loop/files/9166476/Export-20220721T203916Z.zip)

**To Reproduce**
1. Wish I knew..

**Expected behavior**
Persistent therapy settings

**Screenshots**
![DE4373A4-D1C9-4F7B-8471-DF8ACD5E6988](https://user-images.githubusercontent.com/59416392/180407810-1cb65eeb-8bde-4298-8a55-786ecab3f389.jpeg)

**Phone**
 - Hardware: iPhone X
 - OS Version: iOS 15.5

**Loop Version**
 - Version Number: v2.3.0 (57)
 - Repo: LoopKit/Loop, DEV branch

**CGM**
 - Dexcom G6
 - Dexcom App

**Pump**
 - Manufacturer: Omnipod
 - Model: DASH
 - Firmware version: 4.10.0

**Additional context**
Related perhaps to issue #1737 ?

**Comment by itzhakfranco** [comment]
Had the same issue - make sure the health app works properly and allowed to read and write from loop app

**Comment by ChrisVNDL** [comment]
Hi @itzhakfranco 
Health app has all required permissions granted. That shouldn’t be the problem.
In the mean time I noticed this is a known issue that occurs when the Loop app is closed (crashed, terminated or phone rebooted) and a fix has been made available in the  “settings-storage” build. I’ll be testing that soon to see if that works as expected!
Thanks,
Chris

**Comment by ps2** [comment]
Please let me know if you experience this issue with a build after July 13th, (commit b199c78). That is when the settings storage updates landed in dev.

**Comment by vnephologist** [comment]
@ps2 Should settings carry over from previous master builds? I built from dev today to make transition from Eros to DASH and upon loading for the first time, lost all previous settings. I figured this was expected behavior before I saw this issue.

**Comment by ps2** [comment]
Yes, settings should carry over (be migrated). *note* just because you are navigated into the settings review flow does not mean you have lost your settings. If loop thinks your settings are incomplete/invalid in any way, it will route you through the settings review screens. Any settings you previously had should be remembered, and shown as the initial setting when you review each setting.

**Comment by vnephologist** [comment]
Okay, yes, there was nothing at all there on the initial workflow. Let me know if an issue report would provide any useful info.

**Comment by ps2** [comment]
If you notice this happen (redirect to settings review *and* your previous settings are not showing up in the review), then please capture an issue report, and DM me a link to it in zulip.

**Comment by ps2** [comment]
There have been some additional logging statements added to dev, which may be helpful in diagnosing this issue. Please update when you can.

**Comment by ChrisVNDL** [comment]
Ok, sorry it took so long to get back here. We just got back from holiday.

The dev build from 8 days back still lost its setting on application termination. However, I've just built the current dev (6ae754e) to my son's phone and now it seems to be retaining the therapy accross application restarts correctly!

So I'll close this issue with a big thank you to Pete! 
Thanks a lot for all your hard work; it is incredibly appreciated!

Regards,
Chris
